### ISSUE TITLE
Incompatible with bash-completion 1.3 (_init_completion: command not found)

### ISSUE THREAD

**Issue by andrewbranch** [dsifford/yarn-completion#1]
Hi!

First of all, thank you so much for making this! I've been subscribed to https://github.com/yarnpkg/yarn/issues/609 for a while now, so was super excited to see some progress.

After installing, when I type `yarn ` and press tab, I get the error `-bash: _init_completion: command not found`. From my quick research, `_init_completion` is provided by the bash-completion package starting in major version 2, which supports bash@4.1+. I'm on macOS Sierra (upgraded from Yosemite IIRC), and have bash 3.2.57 installed. My bash-completion package is version 1.3, installed from homebrew.

TLDR, I can't use this completion without upgrading bash-completion, which I can't do without upgrading bash itself, which I don't want to do because... it seems scary. 😁

I found another package that had the same issue and seemed to [fix the issue](https://github.com/mobile-shell/mosh/pull/729) and make the completion compatible with both major versions.

I might be able to take a stab at this a little bit later... but let me know if anyone else starts on it first.

**Comment by dsifford** [comment]
Hey @andrewbranch 

Yeah, I forgot to mention that it requires bash 4. You are correct that mac ships with an extremely outdated version of bash.

> which I can't do without upgrading bash itself, which I don't want to do because... it seems scary.

What makes you nervous about doing that? Homebrew makes it pretty simple to do. I did it on my Macbook Pro and I was up and running in a matter of minutes.

Pretty sure all you need to do is...

```sh
$ brew install bash
[ homebrew magic .... ]
$ chsh
Changing shell for <user>.
Password: 
New shell [/bin/bash]: </path/to/bash4>
```
Dead simple!

Anyway, I looked at the workaround that you posted and that seems reasonable. I'll go in and add that now. Should be able to push the fixw within the hour. Thanks 👍

**Comment by andrewbranch** [comment]
I guess I'm just afraid of change 😄 

But in seriousness, I didn't even realize I was using an outdated version of bash, and I do quite a bit with my terminal daily, and this was literally the first thing I've ever run into that was incompatible with my version of bash. After like 7 seconds of reading about upgrading, I saw one comment out of 20 or so that said the upgrade broke npm for them, which was enough to make me think, "I don't really need yarn completions that badly... but I do need to get work done tomorrow instead of debugging my shell for half the day."

I guess I should upgrade at some point, but I'm hesitant that some obscure tool I rely on for some piece of older infrastructure at work will break down, and I won't know it until I'm in a situation where I _really_ need that obscure tool to work.

Anyways, thanks for considering a workaround!

**Comment by dsifford** [comment]
@andrewbranch Actually, I'm not postiive that just adding that workaround will work. I utilize more `bash-completions` functions than that little script, so it's hard to know.

Specifically, the functions that I use are...
- `compgen`
- `_count_args`
- `_filedir`
- `_get_comp_words_by_ref` (this one I believe works).

If I make a new branch with the changes, would you be willing to test on bash 3? That would be super helpful!

--- 

To comment on your thoughts:

>  this was literally the first thing I've ever run into that was incompatible with my version of bash.

I've ran into issues with various readline functions. I can't recall exactly what they were, but collaborators I've worked with in the past couldn't use some of the shell scripts I wrote which was slightly annoying.

> I saw one comment out of 20 or so that said the upgrade broke npm for them

I'm not sure how this would happen by upgrading. They must have either done something wrong or just forgot to log out and log back in.

> I guess I should upgrade at some point, but I'm hesitant that some obscure tool I rely on for some piece of older infrastructure at work will break down, and I won't know it until I'm in a situation where I really need that obscure tool to work.

Not 100% sure, but I think one of the core goals of the bash team is backwards compatibility for that very reason!

**Comment by andrewbranch** [comment]
> If I make a new branch with the changes, would you be willing to test on bash 3? That would be super helpful!

Yeah, totally!

**Comment by dsifford** [comment]
@andrewbranch Are you able to try something real quick?

Looking closer at that workaround you posted, it should be equivalent to just replacing [this line](https://github.com/dsifford/yarn-completion/blob/master/yarn-completion.bash#L507) with the following:

```sh
if declare -F _init_completions >/dev/null 2>&1; then
   _init_completion
fi
```

Basically, it just jumps over it if it isn't available. `_init_completion` is basically just a convenience function to help deal with redirections and other various annoyances which I don't think really matter all that much for this script. 

Anyway, thanks again. Let me know!

**Comment by dsifford** [comment]
Alternative to above is just delete that line altogether.

**Comment by andrewbranch** [comment]
🎉 

<img width="750" alt="it works" src="https://cloud.githubusercontent.com/assets/3277153/25075195/a79dd0ac-22c3-11e7-8222-d11d385ae814.png">

**Comment by dsifford** [comment]
Great! Thanks again for the help!

Pushing the fix now 😄
